examples/type_system/lib/strong_analysis.dart

```diff
@@ -156,3 +156,21 @@ void _miscDeclAnalyzedButNotTested() {
     // #enddocregion generic-type-assignment-implied-cast
   }
 }
+
+// #docregion inference-using-bounds
+class A<X extends A<X>> {}
+
+class B extends A<B> {}
+
+class C extends B {}
+
+void f<X extends A<X>>(X x) {}
+
+void main() {
+  f(B()); // OK.
+  f(C()); // OK. Without using bounds, inference relying on best-effort
+  // approximations would fail after detecting that `C` is not a subtype of `A<C>`.
+  f<B>(C()); // OK.
```

Comment:
The long comment ends up formatting in a way that makes it hard to recognize which statement the 2nd part of it is attached to. How about something like the following?

```suggestion
  f(B()); // OK.

  // OK. Without using bounds, inference relying on best-effort approximations
  // would fail after detecting that `C` is not a subtype of `A<C>`.
  f(C());

  f<B>(C()); // OK.
```

Reply:
Thanks, I didn't like how this ended up but wasn't sure what to do, so this is a big help!